autode/wrappers/QChem.py

```diff
@@ -0,0 +1,502 @@
+              3  H         0.7551329498    0.4500625364    0.0000000000
+            Point Group: cs    Number of degrees of freedom:     3
+            """
+
+            start_idx = i+2
+            end_idx = start_idx + calc.molecule.n_atoms
+
+            atoms = [Atom(_l.split()[1], *_l.split()[2:5])
+                     for _l in calc.output.file_lines[start_idx:end_idx]]
+
+        return atoms
+
+    def get_atomic_charges(self, calc) -> List:
+        raise NotImplementedError
+
+    def get_energy(self, calc) -> float:
+        """Get the total electronic energy from the calculation"""
+
+        for line in reversed(calc.output.file_lines):
+
+            if 'Total energy' in line:
+                try:
+                    return float(line.split()[-1])
+
+                except (TypeError, ValueError, IndexError):
+                    break
+
+        raise CouldNotGetProperty('energy')
+
+    def get_gradients(self, calc) -> np.ndarray:
+        """Gradient of the potential energy"""
+
+        grad = []
+
+        for i, line in enumerate(calc.output.file_lines):
+
+            if 'Cartesian Gradient' not in line:
+                continue
+
+            """e.g.
+            
+                        Cartesian Gradient (au)
+             ATOM              X           Y           Z
+            1  O           0.000005   -0.000002    0.000000
+            2  H           0.000017    0.000001    0.000000
+            3  H          -0.000021    0.000001    0.000000
+            """
+
+            start_idx = i+2
+            end_idx = start_idx+calc.molecule.n_atoms
+
+            grad = [[float(val) for val in _l.split()[2:]]
+                    for _l in calc.output.file_lines[start_idx:end_idx]]
+
+        # Convert from Ha a0^-1 to Ha A-1
+        return np.array(grad) / Constants.a0_to_ang
+
+    def get_hessian(self, calc) -> np.ndarray:
+        """Extract the mass-weighted non projected Hessian matrix
+        NOTE: Required $rem vibman_print 4 $end in the input"""
+        expected_shape = (3 * calc.molecule.n_atoms, 3 * calc.molecule.n_atoms)
+
+        hess = self._extract_mass_weighted_hessian(calc)
+        if hess.shape != expected_shape:
+            raise CouldNotGetProperty('hessian')
+
+        atom_masses = self._extract_atomic_masses(calc)
+
+        # Un-mass weight
+        mass_arr = np.repeat(atom_masses, repeats=3, axis=np.newaxis)
+        hess *= np.sqrt(np.outer(mass_arr, mass_arr))
+
+        # and convert from atomic units (Ha/a0^2) to base units (Ha/Å^2)
+        return hess / Constants.a0_to_ang ** 2
+
+    @staticmethod
+    def _extract_atomic_masses(calc) -> np.ndarray:
+
+        masses = []
+        for line in calc.output.file_lines:
+
+            if 'Has Mass' in line:
+                # e.g.
+                #   Atom    1 Element O  Has Mass   15.99491
+
+                mass = float(line.split()[-1])
+                masses.append(mass)
```

Comment:
again, seems weird to go through the whole lot only to discard most?

Reply:
There should be only one set but I've not seen a good sample of QChem output files so this is a check more than anything...